src/test/java/org/apache/ibatis/submitted/sqlprovider/OurSqlBuilder.java

```diff
@@ -214,4 +219,107 @@ public String buildSelectByIdAndNameMultipleParamAndProviderContext(final Intege
     }}.toString();
   }
 
+  private Class getEntityClass(ProviderContext providerContext){
+    Method mapperMethod = providerContext.getMapperMethod();
+    Class<?> declaringClass = mapperMethod.getDeclaringClass();
+    Class<?> mapperClass = providerContext.getMapperType();
+
+    Type[] types = mapperClass.getGenericInterfaces();
+    for (Type type : types) {
+      if (type instanceof ParameterizedType) {
+        ParameterizedType t = (ParameterizedType) type;
+        if (t.getRawType() == declaringClass || mapperClass.isAssignableFrom((Class<?>) t.getRawType())) {
+          Class<?> returnType = (Class<?>) t.getActualTypeArguments()[0];
+          return returnType;
+        }
+      }
+    }
+    throw new RuntimeException("The interface ["
+        + mapperClass.getCanonicalName() + "] must specify a generic type.");
+  }
+
+  private Map<String, String> getColumnMap(ProviderContext context){
+    Class entityClass = getEntityClass(context);
+    Field[] fields = entityClass.getDeclaredFields();
+    Map<String, String> columnMap = new LinkedHashMap<String, String>();
+    for (Field field : fields) {
+      BaseMapper.Column column = field.getAnnotation(BaseMapper.Column.class);
+      if(column != null){
```

Comment:
Could you apply the formatter, please?
The Eclipse formatter config is [here](https://github.com/hazendaz/build-tools/blob/master/src/main/resources/eclipse-formatter-config-2space.xml). IntelliJ can import it, I think.

Reply:
Formatting has not changed the code I added, it has an impact on the original test code.

For example, the changes shown in the image below, so I didn't format it.
![image](https://user-images.githubusercontent.com/1787798/54476595-07f04280-483a-11e9-84e9-6342a919776c.png)